wallet/chainntfns.go

```diff
@@ -305,6 +307,21 @@ func (w *Wallet) addRelevantTx(dbtx walletdb.ReadWriteTx, rec *wtxmgr.TxRecord,
 		for _, addr := range addrs {
 			ma, err := w.Manager.Address(addrmgrNs, addr)
 			if err == nil {
+				// Prevent addresses from non-default scopes to
+				// be detected here. We don't watch funds sent
+				// to non-default scopes in other places either,
+				// so detecting them here would mean we'd also
+				// not properly detect them as spent later.
+				scopedManager, _, err := w.Manager.AddrAccount(
+					addrmgrNs, addr,
+				)
+				if err != nil {
+					return err
+				}
+				if !waddrmgr.IsDefaultScope(scopedManager.Scope()) {
```

Comment:
So the scenario here is us picking up something like a plain force close that sends to a `p2kh` addr in lnd's bricksquad scope?

Reply:
In this explicit scenario it were P2WKH addresses created from the public keys returned from `walletrpc.DeriveAddress` (derives from the bricksquad scope) which were used in the `bos create-balanced-channel` scenario. And they only showed up as ghost UTXOs if the user also ran a TX reset with the `--reset-wallet-transactions` flag.

Reply:
Force closes of static_remote_key channels wouldn't be affected because even though it's an address derived from the same bricksquad scope, the TX spending from that address again would go into the wallet and be seen as a "relevant TX" while in this scenario it wouldn't as the spending TX was the P2WSH funding TX that the wallet didn't see as its "own".